components/banner/countdown.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component, Fragment } from 'react'
+import classNames from 'classnames'
+import { Button } from 'antd';
+import stylesheet from './countdown.scss'
+import { countdownInfo } from '../constants'
+import { InputForm } from '../inputForm/input-form';
+
+const tokensalePhaseStartDate = countdownInfo.finalDate;//month starts with 0
+const millisecondsInOneDay = 86400000;
+const millisecondsInOneHour = 1000 * 60 * 60;
+const millisecondsInOneMinute = 1000 * 60;
+const millisecondsInOneSeconds = 1000;
+
+class Timer extends Component {
+  constructor(){
+    super();
+
+    this.state = { currentCountdown: this.getCountdown() }
+    this.intervalId = null;
+
+  }
+
+  getCountdown = () => {
+    var today = new Date();
+    var difference = tokensalePhaseStartDate.getTime() - today.getTime();
+
+    if(difference < 0){
+      clearInterval(this.intervalId);
+      return;
+    }
+
+    var daysUntil = Math.floor(difference / millisecondsInOneDay);
+
+    var hourMilliseconds = difference - (daysUntil * millisecondsInOneDay);
+    var hours = Math.floor(hourMilliseconds / millisecondsInOneHour);
+
+    var minutesMilliseconds = difference - (daysUntil * millisecondsInOneDay) - (hours *  millisecondsInOneHour);
+    var minutes = Math.floor(minutesMilliseconds / millisecondsInOneMinute);
+
+    var secondsMillisecons = difference - (daysUntil * millisecondsInOneDay) - (hours * millisecondsInOneHour) - (minutes * millisecondsInOneMinute);
+    var seconds = Math.floor(secondsMillisecons / millisecondsInOneSeconds);
+
+    return {days: daysUntil, hours: hours, minutes: minutes, seconds: seconds};
+  }
+
+  startCountdown = () => {
+    this.setState({currentCountdown: this.getCountdown() });
+    this.intervalId = setInterval(() =>{
+      this.setState({currentCountdown: this.getCountdown() });
+    }, 1000);
+  }
+
+  componentDidMount = () => {
+    this.startCountdown();
+  }
+
+  componentWillUnmount = () =>{
+    clearInterval(this.intervalId);
+  }
+
+  render(){
+    const { day, hour, minut, second } = countdownInfo.parts;
+    const { days, hours, minutes, seconds } = this.state.currentCountdown;
+
+    return (
+      <div className={
+          classNames({
+            'Timer': true
```

Comment:
Why using `classNames` when you are setting the variables to `true`? You can do just do `className='Timer'` directly. I know this code is probably pasted, but I wonder why this wasn't addressed. `classNames` is a small library but each call is still a function call over a `string` definition.

Reply:
Yea it kinda was, I tweaked the already existing component, didn't even notice that. Will change. That is indeed a weird way to use the library.